File: javascript/easy/majorityElement.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let counter = 0, majorityElement = 0;
    for(let num of nums) {
        if(counter == 0) {
            majorityElement = num;
        }

        if(majorityElement == num) {
            counter++;
        } else {
            counter--;
        }
    }
    return majorityElement;
};

File: ruby/easy/majority_element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
    counter, max_element = 0, 0
    nums.each do |num|
        max_element = num if counter == 0
        counter += ((max_element == num) ? 1 : -1)
    end
    return max_element
end
